### ISSUE TITLE
partial match string getting extracted

### ISSUE THREAD

**Issue by vi3k6i5** [vi3k6i5/flashtext#1]
example:

```
from flashtext.keyword import KeywordProcessor
keyword_processor = KeywordProcessor()
keyword_processor.add_keyword('python prog','Python')
print(keyword_processor.extract_keywords('i like python programming'))
```

ourput:
`['Python']`

expected output:
`[]`

**Comment by vi3k6i5** [comment]
Resolved in version: 1.3

**Comment by vi3k6i5** [comment]
Added test case 44 for the same bug.
